katello_certs_tools/rhn_rpm.py

```diff
@@ -75,8 +75,8 @@ def __nonzero__(self):
 
     def checksum_type(self):
         if self.hdr[rpm.RPMTAG_FILEDIGESTALGO] \
-           and self.hdr[rpm.RPMTAG_FILEDIGESTALGO].decode('utf-8') in PGPHASHALGO:
-            checksum_type = PGPHASHALGO[self.hdr[rpm.RPMTAG_FILEDIGESTALGO]]
+           and int(self.hdr[rpm.RPMTAG_FILEDIGESTALGO]) in PGPHASHALGO:
```

Comment:
That's how I read it too, yeah.

Reply:
But if it would be an int, it should have thrown an `AttributeError: 'int' object has no attribute 'decode'`, which I've never seen.

(doesn't increase my liking of this code)

Reply:
this code is unused (and I've removed it now)

explains why it wasn't working but nobody complained ;)

Reply:
Maybe I never looked at it because I looked at #24 to remove the file altogether. It failed but now I finally took the time to make it pass.